lib/ukboot/include/uk/boot.h

```diff
@@ -0,0 +1,44 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+#ifndef __UK_BOOT_BOOT_H__
+#define __UK_BOOT_BOOT_H__
+
+#include <uk/config.h>
+#include <errno.h>
+#include <uk/plat/bootstrap.h>
+
+#if !__ASSEMBLY__
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Requests the "init" thread to perform a shutdown
+ *
+ * @param target
+ *   The action that shall be performed after shutdown:
+ *   - `UKPLAT_HALT`: Terminate/Halt/Power-off
+ *   - `UKPLAT_RESTART`: Restart/Reboot
+ *   - `UKPLAT_CRASH`: Indicate a crash
+ * @returns
+ *   - (0): Success, shutdown is requested.
+ *   - (1): A shutdown is already in progress, note that `request` is
+ *          ignored.
+ *   - (-ENOTSUP): Shutdown requests are not supported.
+ *   - (-EINVAL): Unsupported shutdown `target`.
+ */
+#if CONFIG_LIBUKBOOT_MAINTHREAD
+int uk_boot_shutdown_req(enum ukplat_gstate target);
+#else /* !CONFIG_LIBUKBOOT_MAINTHREAD */
+#define uk_boot_shutdown_req(target) \
+	({ -ENOTSUP; })
+#endif /* !CONFIG_LIBUKBOOT_MAINTHREAD */
+
+#ifdef __cplusplus
+}
+#endif
+#endif /* !__ASSEMBLY__ */
+#endif /* !__UK_BOOT_BOOT_H__ */
```

Comment:
Why `!__UK_BOOT_BOOT_H__`?

Reply:
Because of libukboot and boot.h:
`__UK_LIBNAME_HEADERNAME_H__`

Reply:
Yes I meant why the `!`?

Reply:
:magic_wand: You did not see this 😄